core/src/style/style.h

```diff
@@ -81,31 +63,42 @@ class Style {
     /* Draw mode to pass into <VboMesh>es created with this style */
     GLenum m_drawMode;
 
-    /* Set of strings defining which data layers this style applies to, 
-     * along with the style parameters corresponding to these data layers */
-    std::vector< std::pair<std::string, StyleParams> > m_layers;
-    
+    /* Set of strings defining which data layers this style applies to,
+     * along with the style paramter map corresponding to these data layers,
+     * to be parsed explicitly by styles for their style parameters*/
+    std::vector< std::pair<std::string, StyleParamMap> > m_layers;
+
+    // cache the styleParameters for a layer (later for a ruleID(uniquely generated))
+    // NOTE: no use at present!!!
+    std::unordered_map<std::string, void*> m_styleParamCache;
```

Comment:
While working on the subLayers feature, I though of another approach of caching!!
Instead of caching the "parsed" style properties in individual base styles, I cache the StyleParamMap in the uber style class (static member).

I could do this change here, but it will not be much different to what the present implementation is doing.